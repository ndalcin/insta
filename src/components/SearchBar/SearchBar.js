import React from 'react'

const SearchBar = ({ query, handleInputChange, handleSubmit }) => {
	return (
		<div>
			<input onChange={handleInputChange} value={query} />
			<button onClick={handleSubmit} type="submit">
				search gifs
			</button>
		</div>
	)
}

export default SearchBar
