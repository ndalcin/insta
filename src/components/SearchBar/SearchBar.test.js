import React from 'react'
import { shallow } from 'enzyme'
import SearchBar from './SearchBar'

describe('SearchBar', () => {
	let searchBar

	beforeEach(() => {
		searchBar = shallow(
			<SearchBar
				handleSubmit={jest.fn()}
				handleInputChange={jest.fn()}
				query={''}
			/>
		)
	})

	it('should render correctly', () => {
		expect(searchBar).toMatchSnapshot()
	})
})
