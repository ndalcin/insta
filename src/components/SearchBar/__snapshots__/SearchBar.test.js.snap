// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    handleInputChange={[Function]}
    handleSubmit={[Function]}
    query=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          value=""
/>,
        <button
          onClick={[Function]}
          type="submit"
>
          search gifs
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "search gifs",
          "onClick": [Function],
          "type": "submit",
        },
        "ref": null,
        "rendered": "search gifs",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            value=""
/>,
          <button
            onClick={[Function]}
            type="submit"
>
            search gifs
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "search gifs",
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "search gifs",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
